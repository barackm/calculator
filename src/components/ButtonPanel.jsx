import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = (props) => {
  const handleClick = (buttonName) => {
    props.clickHandler(buttonName); //eslint-disable-line
  };
  const group1 = ['AC', '+/-', '%', 'รท'];
  const group2 = ['7', '8', '9', 'x'];
  const group3 = ['4', '5', '6', '-'];
  const group4 = ['1', '2', '3', '+'];
  const group5 = ['0', '.', '='];
  const operators = ['+', '-', '=', 'รท', 'x'];
  return (
    <>
      <div className="btn-group">
        {group1.map((item) => (
          <Button name={item} key={item} onClick={handleClick} color={!operators.includes(item) ? 'white' : 'orange'} />
        ))}
      </div>
      <div className="btn-group">
        {group2.map((item) => (
          <Button name={item} key={item} onClick={handleClick} color={!operators.includes(item) ? 'white' : 'orange'} />
        ))}
      </div>
      <div className="btn-group">
        {group3.map((item) => (
          <Button name={item} key={item} onClick={handleClick} color={!operators.includes(item) ? 'white' : 'orange'} />
        ))}
      </div>
      <div className="btn-group">
        {group4.map((item) => (
          <Button name={item} key={item} onClick={handleClick} color={!operators.includes(item) ? 'white' : 'orange'} />
        ))}
      </div>
      <div className="btn-group">
        {group5.map((item) => (
          <Button name={item} key={item} onClick={handleClick} wide={item === '0'} color={!operators.includes(item) ? 'white' : 'orange'} />
        ))}
      </div>
    </>
  );
};

ButtonPanel.prototype = {
  clickHandler: PropTypes.func.isRequired,
};
export default ButtonPanel;
