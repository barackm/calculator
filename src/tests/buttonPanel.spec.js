import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';

describe('Button Panel Component', () => {
  it('Should match spanshot', () => {
    const tree = renderer
      .create(<ButtonPanel clickHandler={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
