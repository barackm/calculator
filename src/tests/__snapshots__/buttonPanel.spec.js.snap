// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Panel Component Should match spanshot 1`] = `
Array [
  <div
    className="btn-group"
  >
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>,
  <div
    className="btn-group"
  >
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      x
    </button>
  </div>,
  <div
    className="btn-group"
  >
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="btn-group"
  >
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="btn-group"
  >
    <button
      className="zero"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className=""
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      =
    </button>
  </div>,
]
`;
